pyhoma/client.py

```diff
@@ -84,6 +88,7 @@ def __init__(
         self.password = password
         self.server = server
 
+        self.setup: Setup | None = None
         self.devices: list[Device] = []
         self.gateways: list[Gateway] = []
```

Comment:
I didn’t see this previously, but we should fix it. It’s not correct to assign [] as default value. See https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument

Reply:
Should I set them to None, just like I did for self.setup?

Reply:
Yes have to do it and then set it to empty in the init

Reply:
I am not sure if I understand, since this is already the init method. Do I need to define these variables outside of my init as well? And then just set them in the init?

Reply:
```
def __init__(self, my_array = None):
  if not my_array:
     my_array = []
```

Reply:
But we are not setting them from the `init`? So I am not sure what to do here. Should I declare these fields outside of init and set them to empty in init?

```python3
    def __init__(
        self,
        username: str,
        password: str,
        server: OverkizServer,
        session: ClientSession = None,
    ) -> None:
        """
        Constructor

        :param username: the username for Tahomalink.com
        :param password: the password for Tahomalink.com
        :param server: OverkizServer
        :param session: optional ClientSession
        """

        self.username = username
        self.password = password
        self.server = server

        self.setup: Setup | None = None
        self.devices: list[Device] = []
        self.gateways: list[Gateway] = []
```